packages/adapters-library/src/SQLiteMetadataProvider.ts

```diff
@@ -229,6 +221,57 @@ export class SQLiteMetadataProvider implements IMetadataProvider {
       }
     }
 
-    return Object.values(poolsMap)
+    Object.values(poolsMap).forEach(({ adapterKey, ...pool }) => {
+      adaptersMap.get(adapterKey)!.push(pool)
+    })
+
+    return adaptersMap
+  }
+
+  private adapterKey(protocolId: string, productId: string) {
+    return `${protocolId}#${productId}`
   }
 }
+
+export function buildMetadataProviders(
```

Comment:
Moved these two functions here from `defiProvider`. I've made a small change that allows us to just pass the the filepath and db options from the lambdas. No need to export SQLiteMetadataProvider and instantiate it there anymore.